backend/src/service/reload.service.ts

```diff
@@ -152,6 +171,22 @@ export default class ReloadService {
     this.repositoryService.southItemRepository.updateSouthItem(southItem.id, command);
     const newItem = this.repositoryService.southItemRepository.getSouthItem(southItem.id)!;
     this.oibusEngine.updateItemInSouth(southId, southItem, newItem);
+
+    const oldScanModeId = southItem.scanModeId;
+    const newScanModeId = newItem.scanModeId;
+    const settings = this.repositoryService.southConnectorRepository.getSouthConnector(southId)!;
+
+    if (oldScanModeId != newScanModeId) {
+      this.repositoryService.southCacheRepository.updateCacheScanModeId(southId, newItem.id, oldScanModeId, newScanModeId);
+
+      if (!settings.history.maxInstantPerItem) {
+        const southItems = this.repositoryService.southItemRepository.getSouthItems(southId);
+        const isOldScanModeUnused = !southItems.some(item => item.scanModeId === oldScanModeId);
+        if (isOldScanModeUnused) {
+          this.repositoryService.southCacheRepository.deleteCacheScanMode(southId, oldScanModeId, 'all');
+        }
+      }
+    }
```

Comment:
This should be called before the `this.oibusEngine.updateItemInSouth(southId, southItem, newItem);` line because it may restart right away the item with the old max instant